cishouseholds/derive.py

```diff
@@ -1,11 +1,24 @@
 import re
 from itertools import chain
+from typing import List
 
 from pyspark.ml.feature import Bucketizer
 from pyspark.sql import DataFrame
 from pyspark.sql import functions as F
 
 
+def assign_unique_id_column(df: DataFrame, column_name_to_assign: str, concat_columns: List[str]):
+    """
+    Assign a unique column from concatinating multiple input columns
+    Parameters
+    ----------
+    df
+    column_name_to_assign
+    concat_columns
+    """
+    return df.withColumn(column_name_to_assign, F.concat(*concat_columns))
+
+
 def assign_filename_column(df: DataFrame, column_name_to_assign: str) -> DataFrame:
     """
     Use inbuilt pyspark function to get name of the file used in the current spark task
```

Comment:
could possible partition by the concat columns and add rownumber column

Reply:
agreed. I think concatenating columns might add the chance of not being able to reuse the function in case a non-string column. 

Also the concatenation process must be less efficient than just selecting a list of columns and create window counting function to finally deduplicate with distinct().

Reply:
Might be bit of a hit to processing, but the current approach does give us reproducibility. Seems to work for timestamps anyway, which is good!